libraries/chain/block_header_state.cpp

```diff
@@ -343,6 +343,14 @@ void finish_next(const block_header_state& prev,
          ilog("New finalizer policy becoming active in block ${n}:${id}: ${pol}",
               ("n",block_header::num_from_id(id))("id", id)("pol", *act));
       }
+
+      if (next_header_state.active_proposer_policy != prev.active_proposer_policy) {
```

Comment:
I don't think this check is valid; since `FC_REFLECT`, duplicates shared_ptr data on reading back in. Instead how about comparing the versions instead.